### ISSUE TITLE
ReturnMergePass: mixedScatteredCandidate index-interleave gate misses default-arm-last switch-expression emission

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3020]
## Summary

Follow-up from adversarial review of PR #3003 (fixes #2973). Non-blocking; no regression relative to `main`.

`ReturnMergePass.mixedScatteredCandidate` (`src/ILInspector.Decompiler/Pipeline/Passes/ReturnMergePass.cs:111-112`) gates the result-temp return-tail structuring on the single default-arm `goto` being **index-interleaved** between the conditional predecessors:

```csharp
&& branchPredIndices[0] > conditionalPredIndices.Min()
&& branchPredIndices[0] < conditionalPredIndices.Max()
```

A switch expression whose default-arm `goto` the compiler emits **after** all conditional guards (i.e. `branchPredIndices[0] > conditionalPredIndices.Max()`) will not match, so that shape would remain CS0161 ("not all code paths return a value") rather than being structured.

## Why this is not a regression / not merge-blocking for #3003

- Shapes the gate does not match are left exactly as on `main` (still CS0161, still tracked by `ValidityCoverageReportingTests.DecompilerAssembly_MissingReturnPopulationIsPinned`). The pin recomputes the live CS0161 population all-or-nothing and passes, so no *new* missing-returns are introduced by the PR.
- The concern was raised analytically (Gemini Pro adversarial review); no failing corpus/self-decompilation case was reproduced. It is a completeness limit, not a correctness defect.

## Caution for the fix

The interleave bounds are a **deliberate safety narrowing**: they distinguish a genuine scattered switch-expression dispatch from a contiguous short-circuit `&&`/`||` chain (see the code comment above `mixedScatteredCandidate` and `isolatedSingleGoto`). Naively removing the bounds (as the review suggested) would let the transform fire on short-circuit false-exit merges — an **over-raise** that can change semantics. Any fix must widen the *scattered-dispatch* recognition without admitting contiguous short-circuit chains, and must be paired with a close negative fixture (a short-circuit chain that must NOT be merged).

## Acceptance

- A default-arm-last result-temp switch-expression fixture self-decompiles to valid C# (all paths return).
- A contiguous short-circuit-chain negative fixture is preserved unchanged (no spurious merge).
- No new CS0161 entries; pin stays green.

_Credit: Gemini Pro adversarial review of PR #3003._

**Comment by richlander** [comment]
Closing as **not planned** — no reproduction from compiler output or a real artifact.

## Analysis

The proposed default-arm-last ordering has not been reproduced from compiler-produced IL or a real assembly. The reconciliation for #3003 records the shape as **synthetic IL that cannot be generated from C#**, and a local synthetic probe on current `main` did not reproduce the claimed CS0161 failure — the decompiler retained explicit `goto` control flow and emitted valid C#:

```csharp
if (true) goto IL_0005;
if (false) goto IL_0005;
goto IL_0005;
return 1;
IL_0005:
return 0;
```

Not idiomatic, but valid, and no missing-return.

## Why the gate stays

The interleave bounds in `mixedScatteredCandidate` are a deliberate safety narrowing, not an incidental ordering assumption. A shared return reached by multiple conditional branches can also be the false exit of a contiguous `&&`/`||` chain. Removing or broadly widening the bounds would accept a shape without positive product evidence and make the transform eligible for short-circuit near-misses — an over-raise that can change recompiled semantics.

Current `main` also includes #3063, which strengthened scattered-dispatch recognition in `StructuringPass` without removing the `ReturnMergePass` bounds, preserving the same distinction: scattered switch dispatches require positive structural evidence, while contiguous short-circuit chains stay combined.

## Reopening criteria

A future proposal should begin with a compiler-produced or real-artifact method that:

1. places the single default-arm `goto` after all conditional predecessors;
2. currently decompiles to CS0161 or otherwise invalid C#;
3. demonstrates why existing `StructuringPass` recovery is insufficient; and
4. has a close contiguous short-circuit negative that the new discriminator declines.

If such a witness is found, the fix should add the narrowest new structural proof rather than remove the interleave bounds wholesale.
